src/OrchardCore.Themes/TheAdmin/wwwroot/Scripts/TheAdmin.js

```diff
@@ -7038,21 +7038,37 @@ function confirmDialog(title, message, handler) {
         message = $('#confirmRemoveModalMetadata').data('message');
     }
 
+    if (okText === undefined) {
+        okText = $('#confirmRemoveModalMetadata').data('ok');
+    }
+
+    if (cancelText === undefined) {
+        cancelText = $('#confirmRemoveModalMetadata').data('cancel');
+    }
+
+    if (okCssClass === undefined) {
+        okCssClass = $('#confirmRemoveModalMetadata').data('okClass');
+    }
+
+    if (cancelCssClass === undefined) {
+        cancelCssClass = $('#confirmRemoveModalMetadata').data('cancelClass');
+    }
+
     $(`<div id="confirmRemoveModal" class="modal" tabindex="-1" role="dialog">
```

Comment:
Does this have the potential to be added multiple times to the dom ? 

Multiple calls to confirmDialog() will append multiple modals with the same ID in the dom, making for invalid html.

Reply:
It will not added multiple times, we already discussed this at the time when I introduced the confirm modal